### PR TITLE
Support short options in lsbom.

### PR DESCRIPTION
**Pull request by grp**
Short options can be combined and used as a single flag. For split
short options, the value can follow in a combined option or in the
next option.

### CODE REVIEW

File: Libraries/libbom/Tools/lsbom.cpp [hunk 1]
```diff
@@ -114,67 +114,91 @@ Options::
 {
 }
 
+static ext::optional<std::vector<Options::PrintItem>>
+ParsePrintFormat(std::string const &print)
+{
+    auto format = std::vector<Options::PrintItem>();
+
+    for (char c : print) {
+        switch (c) {
```
Comment (Ktwu) [inline Libraries/libbom/Tools/lsbom.cpp]: Hmm...is it possible you do do something more clever, like create a map for characters -> options, then just reference that?

Comment (grp) [inline Libraries/libbom/Tools/lsbom.cpp]: Yah, although it'd be a few more lines of code and a little slower. :( `std::unordered_map` requires allocation and can't be `constexpr`, so it'd have to allocate memory at least once. Would also take an extra line to search in the map than just print out. A better language would make this easy...

File: Libraries/libbom/Tools/lsbom.cpp [hunk 2]
```diff
@@ -114,67 +114,91 @@ Options::
...
-        if (!result.first) {
-            return result;
-        }
+    if (!arg.empty() && arg[0] == '-') {
+        if (arg == "--help") {
+            return libutil::Options::Current<bool>(&_help, arg);
+        } else if (arg == "--arch") {
+            return libutil::Options::Next<std::string>(&_arch, args, it);
+        } else {
+            std::pair<bool, std::string> result = std::make_pair(true, std::string());
```
Comment (sas) [inline Libraries/libbom/Tools/lsbom.cpp]: Here you can use `auto result = std::make_pair(true, std::string());`.

File: Libraries/libbom/Tools/lsbom.cpp [hunk 3]
```diff
@@ -114,67 +114,91 @@ Options::
 {
 }
 
+static ext::optional<std::vector<Options::PrintItem>>
+ParsePrintFormat(std::string const &print)
+{
+    auto format = std::vector<Options::PrintItem>();
```
Comment (grp) [inline Libraries/libbom/Tools/lsbom.cpp]: 👍

File: Libraries/libbom/Tools/lsbom.cpp [hunk 4]
```diff
@@ -114,67 +114,91 @@ Options::
...
-        return libutil::Options::Current<bool>(&_noModes, arg, it);
-    } else if (arg == "--arch") {
-        return libutil::Options::Next<std::string>(&_arch, args, it);
-    } else if (arg == "-p") {
-        ext::optional<std::string> print;
-        auto result = libutil::Options::Next<std::string>(&print, args, it);
-        if (!result.first) {
-            return result;
-        }
+    if (!arg.empty() && arg[0] == '-') {
```
Comment (grp) [inline Libraries/libbom/Tools/lsbom.cpp]: 👍

**Pull request merged**